algorithms/src/std_algorithms/numeric/Kokkos_ExclusiveScan.hpp

```diff
@@ -275,16 +295,25 @@ OutputIteratorType exclusive_scan_default_op_impl(const std::string& label,
 
   // aliases
   using index_type = typename InputIteratorType::difference_type;
+  using func_type_no_init =
+      ExclusiveScanDefaultFunctorNoInit<ExecutionSpace, index_type, ValueType,
+                                        InputIteratorType, OutputIteratorType>;
   using func_type =
       ExclusiveScanDefaultFunctor<ExecutionSpace, index_type, ValueType,
                                   InputIteratorType, OutputIteratorType>;
 
   // run
   const auto num_elements =
       Kokkos::Experimental::distance(first_from, last_from);
-  ::Kokkos::parallel_scan(label,
-                          RangePolicy<ExecutionSpace>(ex, 0, num_elements),
-                          func_type(init_value, first_from, first_dest));
+  if (init_value == ::Kokkos::reduction_identity<ValueType>::sum())
```

Comment:
instead of this, would it make sense to just add the overloads *without* the init_value? 
It would be inconsistent with the c++ std but I don't see why not doing it since we already have a different API anyway.

Reply:
`std::inclusive_scan` doesn't have an `init_value` (in the overloads without `BinaryOperator`) which makes it difficult to provide extra overloads using neutral elements. In the end, we care more about exclusive scan in `ArborX` anyway so it might be something worth discussing, though.

Reply:
what i mean is that regardless of the init _value, we can detect the value_type and if it is one that has a trivial neutral value (e.g. for float, double, etc) we use the more efficient implementation, if it does not have a trivial neutral element (e.g. user defined type or whatnot) we can use the impl with non-trivial identity that is less efficient

Reply:
I think which impl we use depends on the `value_type` being used not on whether we have an `init_value`